bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/WakeOnLanUtility.java

```diff
@@ -35,32 +35,43 @@
  *
  * @author Arjan Mels - Initial contribution
  * @author Laurent Garnier - Use improvements from the LG webOS binding
+ * @author Nick Waterton - use single ip address as source per interface
  *
  */
 @NonNullByDefault
 public class WakeOnLanUtility {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(WakeOnLanUtility.class);
-    private static final Pattern MAC_REGEX = Pattern.compile("(([0-9a-fA-F]{2}[:-]){5}[0-9a-fA-F]{2})");
     private static final int CMD_TIMEOUT_MS = 1000;
+    private static String host = "";
 
-    private static final String COMMAND;
-    static {
-        String os = System.getProperty("os.name").toLowerCase();
-        LOGGER.debug("os: {}", os);
-        if ((os.contains("win"))) {
-            COMMAND = "arp -a %s";
-        } else if ((os.contains("mac"))) {
-            COMMAND = "arp %s";
-        } else { // linux
-            if (checkIfLinuxCommandExists("arp")) {
+    /**
+     * Get os command to find MAC address
+     *
+     * @param hostName Host Name (or IP address) of host to retrieve command for
+     * @return os COMMAND
+     */
+    public static String getCommand() {
+        String os = System.getProperty("os.name");
+        String COMMAND = "";
+        if (os != null) {
+            os = os.toLowerCase();
+            LOGGER.debug("{}: os: {}", host, os);
+            if ((os.indexOf("win") >= 0)) {
+                COMMAND = "arp -a %s";
+            } else if ((os.indexOf("mac") >= 0)) {
                 COMMAND = "arp %s";
```

Comment:
Just to be sure, arp command is known to be available on all mac hosts?

Reply:
As far as I know, it is.